CONTRIBUTING.md

```diff
@@ -377,9 +363,29 @@ Now code linters and formatters will be run each time you commit changes.
 
 You can skip these checks with `git commit --no-verify` or with the short version `git commit -n`.
 
+### Summary of pre-commit hooks
+
+The following section describes some of the core pre-commit hooks used by the repository.
+See `.pre-commit-config.yaml` for a full list.
+
+C++/CUDA is formatted with [`clang-format`](https://clang.llvm.org/docs/ClangFormat.html).
+
+Additionally, many editors have plugins or extensions that you can set up to automatically run
+`clang-format` either manually or on file save.
```

Comment:
I would move this sentence out of the pre-commit section. It's useful to know and belongs in the contributing guide, but not here. Maybe in a separate second-level header `# Format on save` or something? I think you can implement something similar with black too, or at least I recall some other devs mentioning it (no need to spend time figuring it out if you don't know offhand though).